app/eth2wrap/eth2wrap.go

```diff
@@ -163,8 +163,9 @@ func provide[O any](ctx context.Context, clients []Client,
 	defer cancel()
 
 	var (
-		nokResp forkjoin.Result[Client, O]
-		zero    O
+		nokResp    forkjoin.Result[Client, O]
+		hasNokResp bool
+		zero       O
 	)
 	for res := range join() {
```

Comment:
when the calling context is already/immediately closed, then forkjoin doesn't return any results.